exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillScanRelBase.java

```diff
@@ -26,23 +26,28 @@
 import org.apache.drill.common.exceptions.DrillRuntimeException;
 import org.apache.calcite.rel.core.TableScan;
 import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptCost;
+import org.apache.calcite.plan.RelOptPlanner;
 import org.apache.calcite.plan.RelOptTable;
 import org.apache.calcite.plan.RelTraitSet;
 import org.apache.drill.exec.util.Utilities;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
 
 /**
  * Base class for logical/physical scan rel implemented in Drill.
  */
 public abstract class DrillScanRelBase extends TableScan implements DrillRelNode {
   protected GroupScan groupScan;
   protected final DrillTable drillTable;
+  protected List<SchemaPath> columns;
```

Comment:
The GroupScan interface does not expose getColumns(). As discussed, a better alternative is to modify the interface instead of polluting the DrillScanRelBase class.